api/segments/models.py

```diff
@@ -261,13 +300,44 @@ def get_segment(self):
             rule = rule.rule
         return rule.segment
 
+    def matches_rule(self, rule: "SegmentRule") -> bool:
```

Comment:
Again, let's add a docstring here please. Also, why does this method return a bool, but the parent one doesn't?

Reply:
No problem on the docstring. Added locally. As for returning bool here and not returning a bool in the calling method, it's just about the callers needing a bool or not. The callers of `match_rules_to_segment` don't need a bool as to whether or not there was a match. But it is easy enough to add so I've altered the `match_rules_to_segment` to return a bool.

Reply:
As above, I'd like more of the 'why' here.

Reply:
Ok I've added more to the docstring here as well.